slitlessutils/core/preprocess/crrej/drizzle.py

```diff
@@ -0,0 +1,70 @@
+from drizzlepac import astrodrizzle
+from pathlib import Path
+
+# Args taken from INS HSTaXe FullFrame Cookbooks
+COMMON_ARGS = {
+    'ir': {
+        'mdriztab': True,
+        'preserve': False,
+        'skysub': False,
+        'final_fillval': None
+    },
+    'uvis': {
+        'mdriztab': True,
+        'preserve': False,
+        'skysub': True,
+        'skymethod': 'match',
+        'final_fillval': None
+    },
+    'acs': {
+        'build': True,
+        'mdriztab': True,
+        'in_memory': False,
+        'skysub': False,
+        'driz_separate': False,
+        'median': False,
+        'blot': False,
+        'driz_cr': False,
+        'driz_sep_wcs': False,
+        'driz_combine': True,
+        'final_wcs': False
+    }
+}
+
+
+def drizzle(files, instrument=None, outdir=Path().absolute(), **kwargs):
+    '''
```

Comment:
There are a few things we need to add here.

1) grab the instrument name from the primary header from the first file passed.  This will be tedious as the instruments are indicated with different header keywords.  I suggest the following meta code:
`
with fits.open(FILENAME, mode='readonly') as hdul:
    h = hdul[0].header
    if h['TELESCOP'] == 'HST':
        if h['INSTRUME'] == 'WFC3':
             if h['CSMID'] == 'IR':
                # use WFC3IR settings
             elif h['CSMID'] == 'UVIS':
                # use WFC3UVIS settings
             else: 
                # log a useful warning that CSMID was invalid
       elif h['INSTRUME'] == 'ACS':
           if h['DETECTOR'] == 'WFC':
              # use ACSWFC settings
           elif h['DETECTOR'] == 'SBC':
              # issue warning that SBC does not need drizzling (but we will need to verify this)
       else:
           # issue warning that instrument is not supported and  report the h['INSTRUME'] value

    else: 
      # issue warning that this telescope is not supported and report it
`
That said, I'm not a huge fan of setting these values is a dict in the preamble.  I'm worried that someone will play around with them, and that's far outside the intent.  So, I'd put each of these dicts in their representative space in the meta code.  We will also very likely need to grow this list, and I'm worried this isn't the best way for long-term maintainability. 

This will not catch the instance where someone gives different "types" of images (say a WFC3 and ACS image).  But we'll be able to add that checking later by embedding this inside some other function

2) this method should clean up after itself.  

Last thing.  It should return the file names that it was passed.  This is really minor, but the other function does this, and so it might get frustrating if the APIs aren't the same.

Reply:
Done in https://github.com/spacetelescope/slitlessutils/pull/27/commits/7b102cf9f395e3e0c38ce42c2fe4f40ebf319fdf. I also hid the dictionary args so they aren't user accessible